presto-main/src/main/java/io/prestosql/operator/scalar/LuhnCheckFunction.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator.scalar;
+
+import io.airlift.slice.InvalidUtf8Exception;
+import io.airlift.slice.Slice;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.function.Description;
+import io.prestosql.spi.function.ScalarFunction;
+import io.prestosql.spi.function.SqlType;
+import io.prestosql.spi.type.StandardTypes;
+
+import static io.airlift.slice.SliceUtf8.getCodePointAt;
+import static io.prestosql.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+
+public final class LuhnCheckFunction
+{
+    private LuhnCheckFunction() {}
+
+    @Description("Returns true if the string is a valid credit card number and false if it is not")
```

Comment:
According to https://en.wikipedia.org/wiki/Luhn_algorithm the algorithm is used for more than just credit cards. I think we should also fail (rather than return false) on any non-digit input.

How about this for the description

> Checks that a string of digits is valid according to the Luhn algorithm